content/200-concepts/100-components/02-prisma-client/053-client-extensions/150-type-utilities.mdx

```diff
@@ -23,48 +23,39 @@ The following type utilities are available in Prisma Client:
 - `Result<Type, Arguments, Operation>`: Takes the input arguments and provides the result for a given model and operation. You would usually use this in conjunction with `Args`. As with `Args`, `Result` helps you to re-use existing types to extend or modify them.
 - `Payload<Type, Operation>`: Retrieves the entire structure of the result, as scalars and relations objects for a given model and operation. For example, you can use this to determine which keys are scalars or objects at a type level.
 
-The following example creates a new operation, `customCall`, based on `findFirst`. It has all of the arguments that `findFirst` has, with an additional property, `customProperty`.
+The following example creates a new operation, `exists`, based on `findFirst`. It has all of the arguments that `findFirst`.
 
 ```ts
-const prisma = new PrismaClient()
-
-const xprisma = prisma.$extends({
+const prisma = new PrismaClient().$extends({
   model: {
     $allModels: {
-      // Define a new operation `customCall`.
-      // T corresponds to the current model,
-      // A corresponds to the arguments for the operation.
-      customCall<T, A>(
-        // `this` is the current type (for example
-        // it might be `prisma.user` at runtime).
+      // Define a new `exists` operation on all models
+      // T is a generic type that corresponds to the current model
+      async exists<T>(
+        // `this` refers to the current type, e.g. `prisma.user` at runtime
         this: T,
-        x: Prisma.Exact<
-          A,
-          // For `customCall`, use the arguments from model `T` and the
-          // operation `findFirst`. Add `customProperty` to the operation.
-          Prisma.Args<T, 'findFirst'> & { customProperty: boolean }
-        >
 
-        // Get the correct result types for the model of type `T`,
-        // and the arguments of type `A` for `findFirst`.
-        // `Prisma.Result` computes the result for a given operation
-        // such as `select {id: true}` in function `main` below.
-        //
-      ): Prisma.Result<T, A, 'findFirst'> {
-        // Override type safety here, because we cannot
-        // predict the result types in advance.
-        return {} as any
+        // The `exists` function will use the `where` arguments from the current model, `T`, and the `findFirst` operation
+        where: Prisma.Args<T, 'findFirst'>['where']
+      ): Promise<boolean> {
+        // Retrieve the current model at runtime
+        const context = Prisma.getExtensionContext(this)
+
+        // Prisma Client query that retrieves data based
+        const result = await (context as any).findFirst({ where })
+        return result !== null
       },
     },
   },
```

Comment:
I'm conflicted on whether we should keep this example because it provides extensive usage on how to use TS to build an extension.

Reply:
I'm not sure I follow your concern. I think in-depth examples can be very good and show how our utilities work in a "real world" example.